lib/readyset/query/cached_query.rb

```diff
@@ -0,0 +1,106 @@
+require 'active_model'
+
+require 'readyset/query'
+
+module Readyset
+  module Query
+    # Represents a query that is cached by ReadySet.
+    class CachedQuery
+      include ActiveModel::AttributeMethods
+      include Readyset::Query
+
+      attr_reader :id, :text, :name, :count, :always
+
+      # Returns all of the queries currently cached on ReadySet by invoking the `SHOW CACHES` SQL
+      # extension on ReadySet.
+      #
+      # @return [Array<CachedQuery>]
+      def self.all
+        Readyset.raw_query('SHOW CACHES').map do |result|
+          from_readyset_result(**result.symbolize_keys)
+        end
+      end
+
+      # Drops all the caches that exist on ReadySet.
+      #
+      # @return [void]
+      def self.drop_all!
+        Readyset.raw_query('DROP ALL CACHES')
+
+        nil
+      end
+
+      # Returns the cached query with the given query ID by directly querying ReadySet. If a cached
+      # query with the given ID doesn't exist, this method raises a
+      # `Readyset::Query::NotFoundError`.
+      #
+      # @param [String] id the ID of the query to be searched for
+      # @return [CachedQuery]
+      # @raise [Readyset::Query::NotFoundError] raised if a cached query with the given ID cannot be
+      # found
+      def self.find(id)
+        from_readyset_result(**Query.find_inner('SHOW CACHES WHERE query_id = ?', id))
+      end
+
+      # Constructs a new `CachedQuery` from the given attributes.
+      #
+      # @param [Hash] attributes the attributes from which the `CachedQuery` should be
+      # constructed
+      # @return [CachedQuery]
+      def initialize(id:, text:, name:, always:, count:)
```

Comment:
I differentiate between `.new` and `.from_readyset_result`: `.new` just takes the attributes of the class as params, whereas `from_readyset_result` takes the attributes returned by `SHOW CACHES`.

I did the same thing for `ProxiedQuery`